Source/Initialization/WarpXInitData.cpp

```diff
@@ -231,23 +233,189 @@ WarpX::PostRestart ()
+            }
+            if (E_ext_grid_s == "constant" || E_ext_grid_s == "default") {
+               Efield_aux[lev][i]->setVal(E_external_grid[i]);
+               Efield_cp[lev][i]->setVal(E_external_grid[i]);
+            }
+        }
+        if (B_ext_grid_s == "parse_b_ext_grid_function") {
+
+            InitializeExternalFieldsOnGridUsingParser(Bfield_aux[lev][0].get(),
+                                                      Bfield_aux[lev][1].get(),
+                                                      Bfield_aux[lev][2].get(),
+                                                      str_Bx_ext_grid_function,
+                                                      str_By_ext_grid_function,
+                                                      str_Bz_ext_grid_function,
+                                                      Bx_nodal_flag, By_nodal_flag,
+                                                      Bz_nodal_flag, lev);
+
+            InitializeExternalFieldsOnGridUsingParser(Bfield_cp[lev][0].get(),
+                                                      Bfield_cp[lev][1].get(),
+                                                      Bfield_cp[lev][2].get(),
+                                                      str_Bx_ext_grid_function,
+                                                      str_By_ext_grid_function,
+                                                      str_Bz_ext_grid_function,
+                                                      Bx_nodal_flag, By_nodal_flag,
+                                                      Bz_nodal_flag, lev);
+
+        }
+        if (E_ext_grid_s == "parse_e_ext_grid_function") {
+
+            InitializeExternalFieldsOnGridUsingParser(Efield_aux[lev][0].get(),
+                                                      Efield_aux[lev][1].get(),
+                                                      Efield_aux[lev][2].get(),
+                                                      str_Ex_ext_grid_function,
+                                                      str_Ey_ext_grid_function,
+                                                      str_Ez_ext_grid_function,
+                                                      Ex_nodal_flag, Ey_nodal_flag,
+                                                      Ez_nodal_flag, lev);
+
+            InitializeExternalFieldsOnGridUsingParser(Efield_cp[lev][0].get(),
+                                                      Efield_cp[lev][1].get(),
+                                                      Efield_cp[lev][2].get(),
+                                                      str_Ex_ext_grid_function,
+                                                      str_Ey_ext_grid_function,
+                                                      str_Ez_ext_grid_function,
+                                                      Ex_nodal_flag, Ey_nodal_flag,
+                                                      Ez_nodal_flag, lev);
```

Comment:
All what is done above is copied for E and B. Could this be put in a function call?